Tests/OnlineRetailPortal.Tests/ProductControllerFixture.cs

```diff
@@ -10,72 +11,115 @@ namespace OnlineRetailPortal.Tests
+                Assert.Equal(expectedResponse.Product.Price.IsNegotiable, actualResponse.Product.Price.IsNegotiable);
+                Assert.Equal(expectedResponse.Product.Price.Money.Amount, actualResponse.Product.Price.Money.Amount);
+                Assert.Equal(expectedResponse.Product.Price.Money.Currency, actualResponse.Product.Price.Money.Currency);
+            }
+        }
+        private GetProductResponse GetExpectedProduct()
         {
-            return new GetProductsResponse()
+            return new GetProductResponse()
             {
-                Products = new List<Web.Product>()
+                Product = new Web.Product()
                 {
-                    new Web.Product()
+                    SellerId = "1",
+                    Id = "101",
+                    Name = "Mobile",
+                    Price = new Web.Price
                     {
-                        Id = "103",
-                        Name = "Bottle",
-                        Description = "Green Bottle",
-                        HeroImage =  "example.com" ,
-                        Price = new Web.Price {Money = new Web.Money( 99.99,  "INR" ), IsNegotiable = true },
-                        Category = "Other",
-                        Images = new List<string>() {"ex.com"},
-                        PurchasedDate = DateTime.Now,
-                        PickupAddress = new Web.Address
+                        Money = new Web.Money()
                         {
-                            Line1 = "ABC",
-                            Line2 = "XYZ",
-                            City = "Pune",
-                            Pincode = 411001,
-                            State = "Maharashtra"
-                         }
-
-                    }
-
+                            Amount = 200,
+                            Currency = "INR"
+                        },
+                        IsNegotiable = true
+                    },
+                    Category = "Mobile",
+                    HeroImage = "https://www.olx.in/item/11-pro-max-64-gb-full-box-iid-1540782056/gallery",
+                    Description = "11 pro max 64 gb full box",
+                    Images = new List<string>() { "https://www.olx.in/item/11-pro-max-64-gb-full-box-iid-1540782056/gallery", "https://www.olx.in/item/11-pro-max-64-gb-full-box-iid-1540782056/gallery" },
+                    PickupAddress = new Web.Address { Line1 = "abc", Line2 = "xyz", City = "Pune", State = "Maharashtra", Pincode = 411038 },
+                    PostDateTime = new DateTime(2019, 1, 1),
+                    ExpirationDate = new DateTime(2019, 12, 1).AddDays(30),
+                    PurchasedDate = DateTime.Now,
+                    Status = "Active"
                 }
             };
         }
 
-        private GetProductsServiceRequest GetRequest(int pageNo, int pageSize)
+        [Fact]
+        public async void Test_for_passing_invalid_product_id()
```

Comment:
rename the test case name to something meaningful Test_for_passing_invalid_product_id() does not justify what the test case is trying to test

Reply:
done